libs/execution_context/ebpf_program.c

```diff
@@ -1337,7 +1337,8 @@ ebpf_program_set_tail_call(_In_ const ebpf_program_t* next_program)
         return EBPF_INVALID_ARGUMENT;
     }
 
-    if (state->count == MAX_TAIL_CALL_CNT) {
+    if (state->count == (MAX_TAIL_CALL_CNT - 1)) {
```

Comment:
Any reason for -1? (31)

1. As per doc, Linux support 32 tails calls.
https://www.man7.org/linux/man-pages/man7/bpf-helpers.7.html

2. As per Cilium documentation, support up to 33 tail calls.
  "This limit comes from the 256 bytes per stack frame multiplied by the tail call count limit (33)"

Reply:
The count represents how many tail calls have already been made. I.e. if this is the 31st tail call (i.e. it's already 32 deep) then fail this one.

Reply:
Here is my understanding,
1. 1- 31  is still 31 calls.
2. If let's say, the state starts at 0,  0 - 31, it is  32 calls. Which means 31st call should be allowed.